### DISCUSSION TITLE
Icons doesnt respond to darkmode classes

### DISCUSSION THREAD

**Discussion by Adrian2422** [tailwindlabs/heroicons#913] [Help]
The problem is as follows:
I am using the TailwindCSS library in my React application. 
Icons do not take the colors suitable for dark mode, all the time keeping the color of light mode.
Other elements of application works as intended.

`<SunIcon className="h-6 w-6 dark:text-white text-primary" /> // dark:text-white doesn't work`

**Comment by zefir-git** [comment]
Although I don't use react, I don't think this is a heroicons problem. https://play.tailwindcss.com/O4dIKYGgkM

The icon takes the colour from text colour, i.e. CSS property `color` thanks to using `fill="currentColor"` or `stroke-fill="currentColor"`.

This might be an issue with how you've implemented your dark theme. In any case, here are the docs on TW Dark Mode: https://tailwindcss.com/docs/dark-mode.

When testing your dark mode, it may be helpful to use Firefox Web Developer Tools, which gives you these buttons:
![image](https://user-images.githubusercontent.com/66572326/221121754-abc9b939-cf72-4007-8a32-35ec8b243809.png)
(Inspector tab, right above CSS styles view, right next to the "Filter styles" search). These buttons allow you to change the [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) media feature to see how your theme changes, which is the default strategy of Tailwind which determines whether to use dark mode or not.
